crates/fuzzing/src/oracles/engine.rs

```diff
@@ -93,6 +109,83 @@ pub fn setup_engine_runtimes() {
     crate::oracles::diff_spec::setup_ocaml_runtime();
 }
 
+/// Build a list of allowed values from the given `defaults` using the
+/// `env_list`.
+///
+/// ```
+/// # use wasmtime_fuzzing::oracles::engine::build_allowed_env_list;
+/// // Passing no `env_list` returns the defaults:
+/// assert_eq!(build_allowed_env_list(None, &["a"]), vec!["a"]);
+/// // We can build up a subset of the defaults:
+/// assert_eq!(build_allowed_env_list(Some(vec!["b".to_string()]), &["a","b"]), vec!["b"]);
+/// // Alternately we can subtract from the defaults:
+/// assert_eq!(build_allowed_env_list(Some(vec!["-a".to_string()]), &["a","b"]), vec!["b"]);
+/// ```
+/// ```should_panic
+/// # use wasmtime_fuzzing::oracles::engine::build_allowed_env_list;
+/// // We are not allowed to mix set "addition" and "subtraction"; the following
+/// // will panic:
+/// build_allowed_env_list(Some(vec!["-a".to_string(), "b".to_string()]), &["a", "b"]);
+/// ```
+pub fn build_allowed_env_list(env_list: Option<Vec<String>>, defaults: &[&str]) -> Vec<String> {
```

Comment:
Thanks for the explanation; it took me a second to parse but I think I implemented what you described. I went with the more verbose `if (add_from_defaults && mentioned) || (subtract_from_defaults && !mentioned)` in the logic because I thought that would be the most clear when we read this later...

Reply:
Yeah, I agree, the more verbose condition is a good idea. And more broadly, I think this implementation looks great.